### ISSUE TITLE
Timestamps for persistent timers should be ignored if they are in the future

### ISSUE THREAD

**Issue by keszybz** [systemd/systemd#6821]
```console
$ systemctl --no-pager list-timers
NEXT                          LEFT                   LAST                          PASSED                 UNIT                         ACTIVATES
Fri 2017-09-15 11:05:31 CEST  20h left               Thu 2017-09-14 11:05:31 CEST  3h 1min ago            systemd-tmpfiles-clean.timer systemd-tmpfiles-clean.service
Mon 2018-09-10 00:00:00 CEST  11 months 25 days left Sun 2018-09-09 06:00:00 CEST  11 months 24 days left mlocate-updatedb.timer       mlocate-updatedb.service
Mon 2018-09-10 00:00:00 CEST  11 months 25 days left Sun 2018-09-09 06:00:00 CEST  11 months 24 days left unbound-anchor.timer         unbound-anchor.service

3 timers listed.
Pass --all to see loaded but inactive timers, too.
```
The clock was correctly set at boot, it certainly didn't jump one year forward and then back.
systemd is from recent git.

For reference:
```console
$ systemctl cat mlocate-updatedb.timer
# /usr/lib/systemd/system/mlocate-updatedb.timer
[Unit]
Description=Updates mlocate database every day

[Timer]
OnCalendar=daily
AccuracySec=24h
Persistent=true

[Install]
WantedBy=timers.target
```

**Comment by keszybz** [comment]
```console
$ ls -l /var/lib/systemd/timers/stamp-mlocate-updatedb.timer 
-rw-r--r--. 1 root root 0 Sep  9  2018 /var/lib/systemd/timers/stamp-mlocate-updatedb.timer
```

**Comment by poettering** [comment]
Fix waiting in #6823
